xray/conventions.py

```diff
@@ -826,10 +826,10 @@ def stackable(dim):
         if decode_coords:
             var_attrs = new_vars[k].attrs
             if 'coordinates' in var_attrs:
-                var_coord_names = var_attrs['coordinates'].split()
+                coords = pop_to(var_attrs, new_vars[k].encoding, 'coordinates')
+                var_coord_names = coords.split()
                 if all(k in variables for k in var_coord_names):
                     coord_names.update(var_coord_names)
-                    del var_attrs['coordinates']
```

Comment:
I see.  So you're the approach you're recommending is to only put the coords in `.encoding['coordinates']` if the coordinates are valid.  That is easy enough.